trollmoves/dispatcher.py

```diff
@@ -330,7 +333,10 @@ def create_dest_url(self, msg, client, item):
             if key in mda:
                 mda[key] = aliases.get(mda[key], mda[key])
         path = compose(path, mda)
-        return urljoin(host, path), connection_parameters
+        parts = urlsplit(host)
+        host_path = urlunsplit((parts.scheme, parts.netloc, path,
+                                parts.query, parts.fragment))
+        return host_path, connection_parameters
```

Comment:
does the urlunsplit preserve the username and password ? Maybe add a test for that ?

Reply:
It does. Test added.